invenio_accounts/profiles/schemas.py

```diff
@@ -21,6 +23,25 @@ def validate_visibility(value):
         )
 
 
+def validate_locale(value):
+    """Check if the value is a valid locale."""
+    locales = current_app.extensions["invenio-i18n"].get_locales()
+    locales = [locale.language for locale in locales]
+    locales.append("en_GB")
+    if value not in locales:
+        raise ValidationError(message=str(_("Value must be a valid locale.")))
+    current_app.config["BABEL_DEFAULT_LOCALE"] = value
+    refresh()
+
+
+def validate_timezone(value):
+    """Check if the value is a valid timezone."""
+    if value not in pytz.all_timezones:
+        raise ValidationError(message=str(_("Value must be a valid timezone.")))
```

Comment:
do you need to `cast` to `str`? same above
```suggestion
        raise ValidationError(message=_("Value must be a valid timezone."))
```